### PR TITLE
Limit the number of simultaneous thumbnails that are requested in the itemList.

### PR DESCRIPTION
**Pull request by manthey**
Chrome limits the maximum connectors to a domain to 6.  Girder uses one for getting events, leaving five.  This uses no more than 3 for thumbnails, which leaves some overhead for navigation.  If you navigate between item lists quickly, the connections can still be saturated, as Chrome doesn't cancel the image requests.  We would have to load the image to a hidden element first, and then load it to the visible element to work around that, so that the hidden elements don't vanish on page transitions.

This resolves issue #89.

### CODE REVIEW

**Comment by manthey** [comment]
@dgutman This will substantially improve the browser getting stuck when navigating between folders.  If you find through rapid navigation that this still happens, we can fetch images in a hidden element before showing them so that we track what is happening between navigation.  I don't want to implement that unless it is necessary, as it adds complexity.

**Comment by manthey** [comment]
@zachmullen You are probably the best person to review this, when you get a chance.

**Comment by zachmullen** [comment]
I would expect that deleting the `img` element would automatically cause the request for its data to be canceled, is that not the case?

**Comment by manthey** [comment]
The request is still being processed by the server.  I can't tell if they are still counted by Chrome.

File: web_client/js/itemList.js [hunk 1]
```diff
@@ -1,4 +1,30 @@
...
+     *      thumbnails are located.
+     */
+    function _loadMoreImages(parent) {
+        var loading = $('.large_image_thumbnail img.loading', parent).length;
+        if (maxSimultaneous > loading) {
+          $('.large_image_thumbnail img.waiting', parent).slice(0, maxSimultaneous - loading).each(function () {
+            var img = $(this);
+            img.removeClass('waiting').addClass('loading');
+            img.attr('src', img.attr('deferred-src'));
+          });
```
Comment (zachmullen) [inline web_client/js/itemList.js]: Are we running eslint on this? There are a few style issues here.

**Comment by zachmullen** [comment]
> The request is still being processed by the server. I can't tell if they are still counted by Chrome.

I guess the server wouldn't know until it actually tried to send the bytes back down. This may be something the network tab in chrome can show, in particular if we introduce an artificial delay on the server side.

File: web_client/js/itemList.js (hunk 1 above)
Comment (manthey) [inline web_client/js/itemList.js]: We are running eslint, but oddly, the plugin was given its own .eslintrc file, rather than using Girder's.

Comment (zachmullen) [inline web_client/js/itemList.js]: That's fine. It might be good to enforce a uniform indentation size, but that's out of scope of this issue.

Comment (manthey) [inline web_client/js/itemList.js]: I've fixed this indentation.  I'll make an issue to move to Girder's eslint style, as I don't see any reason not to.

**Comment by manthey** [comment]
I just placed a 30 second delay on all large image thumbnails.   When I navigate between two folders that have large images, the previous pending images remain in Chrome's network list, and DO block things until they resolve.

I can either work around that as an extension to this PR, or only make a separate PR when we decide that it is a real-world problem and not just when I artificially introduce it.

**Comment by zachmullen** [comment]
I think it's fine, mostly just curiosity.

**Pull request merged**